### PR TITLE
Add comment why we define own basic functors

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: src/Generics/SOP/BasicFunctors.hs [hunk 1]
```diff
@@ -9,6 +9,16 @@
 -- library, they're primarily used as parameters for
 -- the 'NP', 'NS', 'POP', and 'SOP' types.
 --
+-- We define own variants of 'Control.Applicative.Const',
+-- 'Data.Functor.Identity.Identity' and 'Data.Functor.Compose.Compose' for
+-- various reasons.
+--
+-- * 'Control.Applicative.Const' and 'Data.Functor.Compose.Compose' become
+-- kind polymorphic only in @base-4.9.0.0@ (@transformeres-0.5.0.0@).
```
Comment (kosmikus) [inline src/Generics/SOP/BasicFunctors.hs]: `transformers`

File: src/Generics/SOP/BasicFunctors.hs [hunk 2]
```diff
@@ -9,6 +9,16 @@
...
+--
+-- * Shorter names are convenient, and pattern synonyms aren't
+-- (yet) powerful enough, particularly exhaustiveness check doesn't work
+-- properly. See <https://ghc.haskell.org/trac/ghc/ticket/8779>.
```
Comment (kosmikus) [inline src/Generics/SOP/BasicFunctors.hs]: Minor coding convention. I prefer multi-paragraph comments to have a final empty comment line `-- `.

**Comment by kosmikus** [comment]
Happy to merge except for the two really minor changes. Thanks for doing the research.

**Comment by phadej** [comment]
amended.

**Comment by kosmikus** [comment]
Thank you.

**Pull request merged**